models/Asset/Service.php

```diff
@@ -336,37 +354,40 @@ public static function rewriteIds($asset, $rewriteConfig)
         return $asset;
     }
 
+
     /**
+     * @internal
+     *
      * @param array $metadata
+     * @param string $mode
      *
      * @return array
      */
-    public static function minimizeMetadata($metadata)
+    public static function minimizeMetadata($metadata, string $mode)
     {
         if (!is_array($metadata)) {
             return $metadata;
         }
 
         $result = [];
         foreach ($metadata as $item) {
-            $type = $item['type'];
-            switch ($type) {
-                case 'document':
-                case 'asset':
-                case 'object':
-                    {
-                        $element = Element\Service::getElementByPath($type, $item['data']);
-                        if ($element) {
-                            $item['data'] = $element->getId();
-                        } else {
-                            $item['data'] = '';
-                        }
-                    }
 
-                    break;
-                default:
-                    //nothing to do
+            $loader = \Pimcore::getContainer()->get('pimcore.implementation_loader.asset.metadata.data');
```

Comment:
metadata grid column update fails for Element types
![image](https://user-images.githubusercontent.com/5137917/88268968-8af99680-ccd3-11ea-9b7e-a2adf3491e59.png)